apps/api/src/services/competition-manager.service.ts

```diff
@@ -190,12 +194,29 @@ export class CompetitionManager {
       // Reset balances
       await this.balanceManager.resetAgentBalances(agentId);
 
-      // Register agent in the competition
+      // Register agent in the competition (automatically sets status to 'active')
       await addAgentToCompetition(competitionId, agentId);
 
-      // Use the agent manager service to reactivate agents - this properly clears the cache
-      await this.agentManager.reactivateAgent(agentId);
-      console.log(`[CompetitionManager] Activated agent: ${agentId}`);
+      // Ensure agent is globally active (but don't force reactivation)
+      const agent = await findAgentById(agentId);
+      if (!agent) {
+        throw new Error(`Agent not found: ${agentId}`);
+      }
+
+      if (agent.status === ACTOR_STATUS.INACTIVE) {
```

Comment:
since agents _will no longer_ be inactivated at the global level after a competition ends, i think this `ACTOR_STATUS.INACTIVE` check isnt needed. we should reject anything that's not in `ACTOR_STATUS.ACTIVE`.